src/test/java/com/googleinterns/zoomtube/utils/TranscriptLineUtilTest.java

```diff
@@ -0,0 +1,103 @@
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.googleinterns.zoomtube.data.TranscriptLine;
+import com.googleinterns.zoomtube.servlets.TranscriptServlet;
+import java.io.IOException;
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import org.w3c.dom.Element;
+
+@RunWith(JUnit4.class)
+public final class TranscriptLineUtilTest {
+  @Rule public final MockitoRule mockito = MockitoJUnit.rule();
+  @Mock private Element node;
+
+  private final LocalServiceTestHelper localServiceHelper =
+      new LocalServiceTestHelper((new LocalDatastoreServiceTestConfig()).setNoStorage(true));
+  private static final String TEST_CONTENT = "test content";
+
+  @Before
+  public void setUp() {
+    localServiceHelper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    localServiceHelper.tearDown();
+  }
+
+  @Test
+  public void createTranscriptLine_transcriptLineSuccessfullyCreated() throws IOException {
+    long lectureId = 1234;
+    Date testDate = new Date();
+    Key testLectureKey = KeyFactory.createKey(LectureUtil.KIND, lectureId);
+    Entity lineEntity = new Entity(TranscriptLineUtil.KIND);
+    lineEntity.setProperty(TranscriptLineUtil.LECTURE, testLectureKey);
+    lineEntity.setProperty(TranscriptLineUtil.CONTENT, TEST_CONTENT);
+    lineEntity.setProperty(TranscriptLineUtil.START, testDate);
+    lineEntity.setProperty(TranscriptLineUtil.DURATION, testDate);
+    lineEntity.setProperty(TranscriptLineUtil.END, testDate);
+
+    TranscriptLine actualLine = TranscriptLineUtil.createTranscriptLine(lineEntity);
+
+    assertThat(actualLine.lectureKey()).isEqualTo(testLectureKey);
+    assertThat(actualLine.start()).isEqualTo(testDate);
+    assertThat(actualLine.duration()).isEqualTo(testDate);
+    assertThat(actualLine.end()).isEqualTo(testDate);
+    assertThat(actualLine.content()).isEqualTo(TEST_CONTENT);
+  }
+
+  @Test
+  public void createEntity_entityAndPropertiesSuccessfullyCreated() throws IOException {
+    Float startDateAsFloat = 23.32F;
+    Float durationAsFloat = 23.32F;
+    Float endDateAsFloat = startDateAsFloat.floatValue() + durationAsFloat.floatValue();
+    long startDateAsLong = (long) 23.32;
+    long durationAsLong = (long) 23.32;
+    long lectureId = 1;
+
+    Entity actualEntity = TranscriptLineUtil.createEntity(
+        lectureId, TEST_CONTENT, startDateAsFloat, durationAsFloat, endDateAsFloat);
+    Key actualKey = KeyFactory.createKey(LectureUtil.KIND, lectureId);
+    Date actualStart = new Date(TimeUnit.SECONDS.toMillis(startDateAsLong));
```

Comment:
These are all part of checking the results so move them to the third section. Also don't forget to declare them only right before you need them.

Reply:
When I said declare them in the right before they are needed I meant:

Key actualKey = KeyFactory.createKey(LectureUtil.KIND, LECTURE_ID_A);
assertThat(actualEntity.getProperty(TranscriptLineUtil.LECTURE)).isEqualTo(actualKey);
assertThat(actualEntity.getProperty(TranscriptLineUtil.CONTENT)).isEqualTo(TEST_CONTENT);
Date actualStart = new Date(TimeUnit.SECONDS.toMillis(startDateAsLong));
assertThat(actualEntity.getProperty(TranscriptLineUtil.START)).isEqualTo(actualStart);
...

Remember this style guide recommendation: https://google.github.io/styleguide/javaguide.html#s4.8.2-variable-declarations

Reply:
Sorry I think I skimmed over the last sentence.

Done!